src/deltacode/utils.py

```diff
@@ -179,40 +191,36 @@ def deltas(deltacode, all_delta_types=False):
     """
     for delta in deltacode.deltas:
         if all_delta_types is True:
-            yield delta.to_dict()
+            yield delta.to_dict(deltacode)
         elif not delta.is_unmodified():
-            yield delta.to_dict()
+            yield delta.to_dict(deltacode)
 
 def calculate_percent(value, total):
     """
     Return the rounded value percentage of total.
     """
-    ratio = (value / total) * 100
-    return round(ratio, 2)
+    try :
+        ratio = (value / total) * 100
+        return round(ratio, 2)
+    except ZeroDivisionError:
+        return 0
 
 class AlignmentException(Exception):
     """
     Named exception for alignment errors.
     """
     pass
 
-def align_trees(a_files, b_files):
-    """
-    Given two sequences of File objects 'a' and 'b', return a tuple of
-    two integers that represent the number path segments to remove
-    respectively from a File path in 'a' or a File path in 'b' to obtain the
-    equal paths for two files that are the same in 'a' and 'b'.
-    """
-    # we need to find one uniquly named file that exists in 'a' and 'b'.
+def align_trees(codebase1, codebase2):
     a_names = defaultdict(list)
-    for a_file in a_files:
-        a_names[a_file.name].append(a_file)
-    a_uniques = {k: v[0] for k, v in a_names.items() if len(v) == 1}
+    for resource in codebase1.walk():
+        a_names[resource.name].append(resource)
+    a_uniques = {k : v[0] for k, v in a_names.items() if len(v) == 1}
 
     b_names = defaultdict(list)
-    for b_file in b_files:
-        b_names[b_file.name].append(b_file)
-    b_uniques = {k: v[0] for k, v in b_names.items() if len(v) == 1}
+    for resource in codebase2.walk():
+        b_names[resource.name].append(resource)
+    b_uniques = {k : v[0] for k, v in b_names.items() if len(v) == 1}
```

Comment:
Like mentioned above, I think we can safely ignore all the logic around aligning trees and storing old_paths etc.